### PR TITLE
bug: resolve type conversion alerts

### PR DESCRIPTION
**Pull request by dduzgun-security**
Resolves `Incorrect conversion between integer types` alerts to catch potential overflows.

### CODE REVIEW

File: command/agent/http.go
Comment (dduzgun-security) [reply command/agent/http.go:918]: Good catch! Silly mistake in the condition 🤦‍♂️ 
This commit should fix it https://github.com/hashicorp/nomad/pull/20553/commits/63923e2f3bd11b33859a2637389ecc9281f7f8ae


```bash
curl -v "http://localhost:4646/v1/jobs?stale=random"
*   Trying [::1]:4646...
* Connected to localhost (::1) port 4646
> GET /v1/jobs?stale=random HTTP/1.1
> Host: localhost:4646
> User-Agent: curl/8.4.0
> Accept: */*
> 
< HTTP/1.1 400 Bad Request
< Vary: Accept-Encoding
< Date: Fri, 10 May 2024 19:48:40 GMT
< Content-Length: 59
< Content-Type: text/plain; charset=utf-8
< 
* Connection #0 to host localhost left intact
Expect `true` or `false` for `stale` query string parameter% 
```

```bash
curl -v "http://localhost:4646/v1/jobs?per_page=random"
*   Trying [::1]:4646...
* Connected to localhost (::1) port 4646
> GET /v1/jobs?per_page=random HTTP/1.1
> Host: localhost:4646
> User-Agent: curl/8.4.0
> Accept: */*
> 
< HTTP/1.1 400 Bad Request
< Vary: Accept-Encoding
< Date: Fri, 10 May 2024 19:49:07 GMT
< Content-Length: 53
< Content-Type: text/plain; charset=utf-8
< 
* Connection #0 to host localhost left intact
Expect a number for `per_page` query string parameter% 
```

**Review by tgross — DISMISSED** [review]
LGTM! Once CI is green we can get this merged and backported.

File: client/lib/numalib/detect_linux.go
Comment (shoenig) [inline client/lib/numalib/detect_linux.go:95]: the error is ignored in the parse but then we check it's not nil? this error has nothing to do with the parsing

File: api/api.go
Comment (shoenig) [inline api/api.go:1187]: math.MaxInt64 is an untyped constant, do we need a cast here?

File: client/lib/numalib/detect_linux.go
Comment (tgross) [reply client/lib/numalib/detect_linux.go]: Oops, fixed that too.

File: api/api.go
Comment (dduzgun-security) [reply api/api.go:1187]: > I think it's because this check is to see if last (which is a uint64) is greater than time.Duration (which is int64)
https://github.com/hashicorp/nomad-enterprise/pull/1433#discussion_r1583156618

Because we do `q.LastContact = time.Duration(last) * time.Millisecond` on the next line.

Comment (tgross) [reply api/api.go]: Right, but I think @shoenig's point is that the `math.MaxInt64` doesn't need the cast here. It compiles without it.

Fixed.

**Pull request merged**
